### ISSUE TITLE
Evaluation procedure

### ISSUE THREAD

**Issue by drprojects** [vLAR-group/GrowSP#6]
Hi, thanks for your interesting work and for sharing your code.

I just read your paper and have a question regarding the evaluation procedure.

The paper explains in _3.4. Implementation_ that the evaluation procedure relies on K-means with K=C the number of classes, followed by a Hungarian Matching step. So far this is clear to me. In section _4.5. Comparison with Self-supervised Learning_, the paper proposes to compare the expressivity of the learnt point features with the features from other self-supervized learning methods. Two evaluation procedures are proposed: the K-Means+Hungarian Matching, and linear probing. This sounds clear to me too. However, when comparing S3DIS Area 5 performance communicated in Table 1 and Table 8, I am inclined to believe that the evaluation procedure used in Table 1 is actually linear probing.

Can you please clarify this for me ? If I am mistaken, what would be the reason for the performance drop of your method with K-means+Hungarian Matching in Table 8 ?

Thanks in advance !

**Comment by zhang-zihui** [comment]
Hi @drprojects, for a fair comparison with CSC and PC, the models used in Table 8 are trained on ScanNet training set, which is different from Table 1.

**Comment by drprojects** [comment]
Hi @zhang-zihui, I better understand the results in Table 8, thank you for the clarification.

If I may, I take this opportunity for another question, regarding the loss function 😇 

In section _3.3. Semantic Primitive Clustering_, in the _Loss Function_ paragraph, I do understand the point-wise pseudo-labels construction:

> Naturally, each superpoint and individual 3D points within it will be given an S-dimensional one-hot pseudo-label.

However, I am not sure I understand the logits construction: 

> For all S primitives, we use the corresponding centroids (PFH simply dropped) estimated by K-means as a classifier to classify all individual 3D points, obtaining S-dimensional logits.

How exactly do you build a point classifier from the S centroids ? How does this produce point-wise logits (which I assume to be a vector of pseudo-probabilities of size S) ?

Thanks in advance for the help !

Best, 
Damien

**Comment by zhang-zihui** [comment]
we compute the cosine similarity of point-wise features and centroids, the centroids here act like a classifier.   The logits are cosine distances after softmax.

**Comment by drprojects** [comment]
I see, thank you for the clarification.
